src/loader.js

```diff
@@ -32,6 +62,9 @@ export function pitch(request) {
     let resultSource;
     if (query.remove) {
       resultSource = '// removed by extract-text-webpack-plugin';
+      if (process.env.NODE_ENV === 'development') {
```

Comment:
Let's use `hot` option for this :+1:

Reply:
`options.hmr || options.hot` or `webpack >= v4.0.0` also adds `this.hot` to the `loaderContext` now

Reply:
`webpack` loader/plugins shouldn't directly rely on `process.env` whenever possible

Reply:
@michael-ciniawsky When I run `../node_modules/.bin/webpack-dev-server --watch --hot` both `query` and `this.options` are missing either "hot" or "hmr". the "this.hot" is also undefined.

Are you sure this is the way?

Reply:
Oh, I get it, you've decided to make it an option? @evilebottnawi told me before it should a default